crates/bevy_transform/src/components/transform.rs

```diff
@@ -222,11 +222,8 @@ impl Transform {
 
     /// Returns a [`Vec3`] of this [`Transform`] applied to `value`.
     #[inline]
-    pub fn mul_vec3(&self, mut value: Vec3) -> Vec3 {
-        value = self.scale * value;
-        value = self.rotation * value;
-        value += self.translation;
-        value
+    pub fn mul_vec3(&self, value: Vec3) -> Vec3 {
+        self.rotation * self.scale * value + self.translation
```

Comment:
```suggestion
        self.scale * self.rotation * value + self.translation
```
Let's keep the ordering consistent here.

Reply:
I'm not sure those are commutatives. I added parentheses to make sur it happens in the right order

Reply:
Is this change necessary (turning the 3 lines into 1)? I would think the compiler optimises it to the same result, and the existing notation is a lot clearer about the order of operations imo.

Though mockersf is right, it definitely should be scale applied to value first, then multiplied by rotation, then lastly translation added. perhaps `translation + rotation * (scale * value)` is nicer?

Reply:
I like to be clear that we're following SRT here: it's much cleaner at a glance if we're adding `Translation` last.

Reply:
It's a matter of perspective I guess;  If you were to consider Scale, Rotation, Translation as independent transformations that you compose like
```
translate(rotate(scale(value)))
```
Then it's a lot more visually similar imo, since the order of operations is 
```
translation + (rotation * (scale * value))
```
It only looks inside out compared to the "SRT" acronym because the inner-most (farthest right) function is applied to the value first